packages/block-library/src/navigation/index.php

```diff
@@ -1512,6 +1512,14 @@ function block_core_navigation_set_ignored_hooked_blocks_metadata( $inner_blocks
  * @return stdClass The updated post object.
  */
 function block_core_navigation_update_ignore_hooked_blocks_meta( $post ) {
+	/*
+	 * In this scenario the user has likely tried to create a navigation via the REST API.
+	 * In which case we won't have a post ID to work with and store meta against.
+	 */
+	if ( ! isset( $post->ID ) ) {
```

Comment:
```suggestion
	if ( empty( $post->ID ) ) {
```
`$post->ID` could be set and contain a falsy value. Then `get_post()` will return the current global post which can have unexpected behavior IMHO.

Reply:
Ah good catch, thanks!